src/main/resources/static/js/userSubjects.js

```diff
@@ -0,0 +1,46 @@
+$(document).ready(function () {
+    $('.add-grade-btn').click(function () {
+        const gradeInput = $(this).closest('tr').find('input[name="grade"]');
+        const gradeTypeSelect = $(this).closest('tr').find('select[name="gradeType"]');
+        const userSubjectId = $(this).closest('tr').find('.remove-btn').data('id');
+        const gradeValue = gradeInput.val();
+        const gradeTypeId = gradeTypeSelect.val();
+        const gradeTypeName = gradeTypeSelect.find("option:selected").text();
+
+        if (!gradeValue || gradeValue < 1 || gradeValue > 6) {
+            alert('Please select a valid grade between 1 and 6.');
+            return;
+        }
+
+        const gradeData = {
+            userSubjectId: userSubjectId,
+            gradeTypeId: gradeTypeId,
+            gradeTypeName: gradeTypeName,
+            gradeValue: gradeValue
+        };
+
+        console.log("Grade data:", gradeData);
```

Comment:
why the log?

Reply:
it was add for debugging purposes to see if  the data being passed around correctly. Shall I remove it?

Reply:
yes. console.logs should never remain in the code